http_client.go

```diff
@@ -110,50 +114,60 @@ func (c *httpClient) Delete(url string, headers http.Header) (Response, error) {
 
 	request.Header = headers
 
-	return c.do(request)
+	res, err := c.Do(request)
+	return toHeimdallResponse(res, err)
 }
 
-func (c *httpClient) do(request *http.Request) (Response, error) {
-	hr := Response{}
+func toHeimdallResponse(response *http.Response, err error) (Response, error) {
```

Comment:
I think we should get rid of converting to heimdall Response and instead return *http.Response back